ystia/yorc
# Feature/gh 553 event store

# Pull Request description

Refactor events package to allow to use new implementations for event store.

## Description of the change

### What I did

- Events package (remaining code about events) refactoring

- Refactor upgrade processes to run them concurrently for all deployments
- Refactor upgrade pre31 process to run it concurrently for all events
- Add config property "migrate_data_from_consul" for store of type Log and Event to ensure migration of logs and events from Consul for at the store initialization
 

### How to verify it

- Add in Yorc config file the lines:

```

  "storage": {
    "reset": true,
    "stores": [
      {
        "name": "myFileStore",
        "migrate_data_from_consul": true,
        "implementation": "fileCache",
        "types":  ["Log", "Event"],
      }
    ]
  }

```

- Check correct config is saved in Consul
- Deploy an app with the related yorc a4c plugin https://github.com/alien4cloud/alien4cloud-yorc-provider/pull/15)
- Check events are stored in work/store directory in clear.
- Check events are correctly polled from A4C and Yorc CLI

If you first run a deployment with default config (Log and event stores implementation is Consul), then restart Yorc with the config above and a reset to true, you can see existing logs and events migrating from Consul to your store.

### Description for the changelog
* Refactor deployments events to allow to use different backends (GH-553](https://github.com/ystia/yorc/issues/553))
## Applicable Issues
#553
